### DISCUSSION TITLE
lockFileMaintenance weekly PR has stale changes

### DISCUSSION THREAD

**Discussion by ctz** [renovatebot/renovate#30088] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

We are using forking renovate to update our rust cargo dependencies.

`lockFileMaintenance` makes the update PR as expected on Monday morning, but the actual commit in the PR is not updated until someone clicks the "If you want to rebase/retry this PR, check this box" box.

Examples:

This week: https://github.com/rustls/rustls/pull/2035 (note commit is two weeks old)
Last week: https://github.com/rustls/rustls/pull/2020 (commit was one week old, box was checked to update it)
Week before: https://github.com/rustls/rustls/pull/1996 (PR 10th june, commit from 5th June)

Our configuration:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "labels": [
    "dependencies"
  ],
  "lockFileMaintenance": {
    "enabled": true
  },
  "prCreation": "not-pending",
  "rangeStrategy": "update-lockfile",
  "stabilityDays": 3,
  "github-actions": {
    "fileMatch": [
      "^ci\\/.*/[^/]+\\.ya?ml$"
    ]
  },
  "packageRules": [
    {
      "matchManagers": [
        "cargo"
      ],
      "matchUpdateTypes": [
        "patch"
      ],
      "enabled": false
    },
    {
      "matchManagers": [
        "cargo"
      ],
      "matchUpdateTypes": [
        "minor"
      ],
      "matchCurrentVersion": "!/^0/",
      "enabled": false
    }
  ]
}
```

**Comment by RahulGautamSingh** [comment, marked as answer]
Not a bug. This is happening due to some of the config settings and possibly your repo's branch protection rules.

**Delay reason**
You have `prCreation=not-pending` which means prs will be created only after all branch status checks pass. One of those check is the `statbility-days` check and since you have set it to 3 the prs will be created 3 days after branch has been created incase the branch was created on the release date of the new version.

**No rebase reason**
You have not set `rebaseWhen` so the default `auto` setting will be used.
_From the docs_
> auto: Renovate will autodetect the best setting. It will use behind-base-branch if configured to automerge or repository has been set to require PRs to be up to date. Otherwise, conflicted will be used instead

ie. unless the branch is conflicted it won't be rebased by itself. 

You can fix this by changing your repo's branch protection rules to require that PRs must be up-to-date before merging. (the bot will detect this setting and rebase PRs by itself)

Or, you can set `rebaseWhen` to `behind-base-branch`

**Comment by cpu** [comment]
Thank you for your help/pointers. What you're describing makes sense :+1:
